### DISCUSSION TITLE
How to wait for an event to be processed?

### DISCUSSION THREAD

**Discussion by nkosi23** [JasperFx/marten#2599] [Q&A]
Thanks for this awesome library, I am getting started with event sourcing and want to use MartenDB along with Wolverine. 

Since my backends are typically called by a single page application, there is a need to wait until commands have been processed before updating the UI. Moreover in situations where entities are created, there is a need to wait until the Id of the entity is returned to know which entity to load (and even if tricks were used to generate the Id before events are even published, we do not want the user to try to navigate to the page before the entity has actually been created otherwise he'll see a 404).

So my question is, when publishing an event using MartenDB, is there a way to await until the message has been processed by at least one consumer? (or is there another approach that would achieve a similar outcome?)

Thank you

**Comment by gfoidl** [comment]
> publishing an event using MartenDB

What do you understand by "publishing an event"?
When working with `session.Events`, etc. you can `await` the operation. I'm not clear if that's what you meant.

> need to wait until commands have been processed before updating the UI

I don't know your architecture / setup / whatever. But a differnet option
* return an identifier to the UI as soon as possible
* use SignalR to send additional information once the asynchronous operation (here not the `async` keyword, but async in messaging terms) is complete

I know that's a quite generic answer, as I don't understand what you're asking for...
